src/aggregate/aggregate_exec.c

```diff
@@ -128,6 +128,14 @@ static size_t serializeResult(AREQ *req, RedisModuleCtx *outctx, const SearchRes
         count++;
         const RLookupKey *rlk = RLookup_GetKey(cv->lastLk, req->requiredFields[currentField], 0);
         const RSValue *v = getReplyKey(rlk, r);
+        // align field value with its type
+        if (rlk->fieldtype == RLOOKUP_C_DBL && v->t != RSVALTYPE_DOUBLE) {
+          double d;
+          RSValue rsv;
+          RSValue_ToNumber(v, &d);
+          RSValue_SetNumber(&rsv, d);
+          v = &rsv;
```

Comment:
You are pointing to a location on the stack that might be freed when you exit the if scope.. get it outside of the if